src/login/logind-dbus.c

```diff
@@ -2038,6 +2044,68 @@ static usec_t nologin_timeout_usec(usec_t elapse) {
         return LESS_BY(elapse, 5 * USEC_PER_MINUTE);
 }
 
+int manager_load_scheduled_shutdown(Manager *m) {
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *usec = NULL,
+               *warn_wall = NULL,
+               *mode = NULL,
+               *wall_message = NULL,
+               *uid = NULL,
+               *tty = NULL;
+        _cleanup_free_ char *unescaped = NULL;
+        int r;
+
+        r = parse_env_file(f, SHUTDOWN_SCHEDULE_FILE,
+                        "USEC", &usec,
+                        "WARN_WALL", &warn_wall,
+                        "MODE", &mode,
+                        "WALL_MESSAGE", &wall_message,
+                        "UID", &uid,
+                        "TTY", &tty);
+        if (r == -ENOENT)
+                return 0;
+        if (r < 0) {
+                log_debug_errno(r, "Failed to parse " SHUTDOWN_SCHEDULE_FILE ": %m");
+                goto out;
+        }
+
+        r = deserialize_usec(strnull(usec), &m->scheduled_shutdown_timeout);
+        if (r < 0)
+                goto out; /* without proper time no point handling the rest */
+
+        HandleAction handle = handle_action_from_string(mode);
+        if (handle != _HANDLE_ACTION_INVALID)
+                m->scheduled_shutdown_type = manager_item_for_handle(handle);
+        else {
+                log_debug("Failed to parse scheduled shutdown type '%s', ignoring.", mode);
+                m->scheduled_shutdown_timeout = 0;
+                r = -EINVAL;
+                goto out;
+        }
+
+        /* remaining settings are not crucial, so ignore errors */
+
+        if (warn_wall && parse_boolean(warn_wall) >= 0)
```

Comment:
these days we try hard not to ignore errors entirely, but do a debug log about them whenever something fails, i.e. change this to:

```c
if (warn_wall) {
        r = parse_boolean(warn_wall);
        if (r < 0) {
                log_debug_errno(r, "Failed to parse xyz field: %m");
                goto finish;
        }

Reply:
ok if we bail out at those places I'd reset the scheduled shutdown completely in the fail path. So we don't half apply the file.